mlflow/utils/credentials.py

```diff
@@ -41,3 +46,129 @@ def read_mlflow_creds() -> MlflowCreds:
+
+    Args:
+        backend: string, the backend of the tracking server. Currently only "databricks" is
+            supported.
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+
+        mlflow.login()
+        with mlflow.start_run() as run:
+            mlflow.log_param("p", 0)
+    """
+    if backend == "databricks":
+        _databricks_login()
+    else:
+        raise ValueError(
+            f"Currently only 'databricks' backend is supported, received `backend={backend}`."
+        )
+
+
+def _check_databricks_auth():
+    # Check if databricks credentials are set.
+    from databricks.sdk import WorkspaceClient
+
+    try:
+        # If credentials are set correctly, `WorkspaceClient()` will not throw an error.
+        WorkspaceClient()
+        return True
+    except Exception:
+        return False
+
+
+def _overwrite_or_create_databricks_profile(
+    file_name,
+    profile,
+    profile_name="DEFAULT",
+):
+    """Overwrite or create a profile in the databricks config file.
+
+    Args:
+        file_name: string, the file name of the databricks config file, usually `~/.databrickscfg`.
+        profile: dict, contains the authentiacation profile information.
+        profile_name: string, the name of the profile to be overwritten or created.
+    """
+    profile_name = f"[{profile_name}]"
+    with open(file_name) as file:
+        lines = file.readlines()
+    start_index = -1
+    end_index = -1
+    # Find the start and end indices of the profile to overwrite.
+    for i in range(len(lines)):
+        if lines[i].strip() == profile_name:
+            start_index = i
+            break
+
+    if start_index != -1:
+        for i in range(start_index + 1, len(lines)):
+            # Reach an empty line or a new profile.
+            if lines[i].strip() == "" or lines[i].startswith("["):
+                end_index = i
+                break
+        end_index = end_index if end_index != -1 else len(lines)
+        del lines[start_index : end_index + 1]
+
+    # Write the new profile to the end of the file.
+    lines.append(profile_name + "\n")
+    lines.append(f"host = {profile['host']}\n")
+    if "token" in profile:
+        lines.append(f"token = {profile['token']}\n")
+    else:
+        lines.append(f"username = {profile['username']}\n")
+        lines.append(f"password = {profile['password']}\n")
+
+    # Write back the modified lines to the file.
+    with open(file_name, "w") as file:
+        file.writelines(lines)
+
+
+def _databricks_login():
+    """Set up databricks authentication and connect MLflow to Databricks tracking server."""
+    if _check_databricks_auth():
+        # Check if the auth has already been set.
+        _logger.info("Succesfully signed in Databricks!")
+        return
+
+    while True:
+        host = getpass.getpass("Databricks Host (should begin with https://): ")
```

Comment:
```suggestion
        host = input("Databricks Host (should begin with https://): ")
```

I think we can use `input` for `host` and `username`.

Reply:
`databricks configure` doesn't mask `host`.

Reply:
sg!